browser/brave_rewards/rewards_tab_helper.cc

```diff
@@ -170,13 +187,51 @@ void RewardsTabHelper::OnRewardsInitialized(RewardsService* rewards_service) {
   }
 }
 
+void RewardsTabHelper::ShouldDetectCreator(
+    ShouldDetectCreatorCallback callback) {
+#if BUILDFLAG(IS_ANDROID)
+  std::move(callback).Run(false);
+#else
+  std::move(callback).Run(rewards_service_ &&
```

Comment:
Any chance to pass it via web_preferences/renderer_preferences to avoid a dedicated async mojo call?

Reply:
Since we already add fields to `web_preferences`, I added a new field `rewards_enabled` there. Let me know what you think.

Reply:
looks good to me.